project/apps/collaboration-service/src/adapters/db/collaboration.supabase.ts

```diff
@@ -93,21 +99,49 @@ export class CollaborationSupabase implements CollaborationRepository {
     return data;
   }
 
-  /**
-   * Retrieves all collaborations based on the provided filters.
-   *
-   * @param {CollabFiltersDto} filters - The filters to apply when fetching collaborations.
-   * @returns {Promise<CollabDto[]>} A promise that resolves to an array of collaboration DTOs.
-   * @throws Will throw an error if the collaborations cannot be fetched.
-   */
-  async findAll(filters: CollabFiltersDto): Promise<CollabDto[]> {
-    const { user_id, collab_user_id, includeEnded, offset, limit, sort } =
-      filters;
+  async findAll(filters: CollabFiltersDto): Promise<CollabCollectionDto> {
+    const {
+      user_id,
+      collab_user_id,
+      has_ended,
+      offset,
+      limit,
+      sort,
+      q_title,
+      q_category,
+      q_complexity,
+    } = filters;
+
+    let filteredQuestionIds = null;
+    if (q_title || q_category || q_complexity) {
+      // query for questions that match the filters
+      const questionFilters: QuestionFiltersDto = {
+        title: q_title,
+        categories: q_category,
+        complexities: q_complexity,
+        includeDeleted: true,
+      };
+
+      // get the filtered question ids
+      // we have to call the question service to get the filtered questions as we don't have direct access to the questions
+      const filteredQuestions = await firstValueFrom(
+        this.questionServiceClient.send<QuestionCollectionDto>(
+          { cmd: 'get_questions' },
+          questionFilters,
+        ),
+      );
+
+      filteredQuestionIds = filteredQuestions.questions.map((q) => q.id);
+    }
 
     let queryBuilder = this.supabase
       .from(this.COLLABORATION_TABLE)
       .select('*', { count: 'exact' });
 
+    if (filteredQuestionIds) {
+      queryBuilder = queryBuilder.in('question_id', filteredQuestionIds);
+    }
+
     // Apply user filtering
     if (user_id && collab_user_id) {
       // Both user_id and collab_user_id are provided
```

Comment:
Optional: should we consider allowing search by username of collab peer on the history given that this is currently implemented as a filter field?

If difficult to implement, perhaps can remove this

Reply:
Not the most straightforward to implement, bcos we cant easily join the `profile` and `collaboration` tables.
Will do this in the future if there time 👍